Trees/DepthFirstSearch.js

```diff
@@ -1,76 +1,213 @@
 /*
- * Author: Surendra Kumar
+ * Author: Sundaram Kumar Jha
  * DFS Algorithm implementation in JavaScript
  * DFS Algorithm for traversing or searching graph data structures.
 */
 
-function traverseDFS (root) {
-  const stack = [root]
-  const res = []
+function dfs(tree) {
+  if (!tree) return;
+  console.log(tree.value);
+  for (const child of tree.children) {
+    dfs(child);
+  }
+}
 
-  while (stack.length) {
-    const curr = stack.pop()
-    res.push(curr.key)
+// This function takes a tree node as an argument, and uses recursion to traverse 
+// the tree in a depth-first manner. The function first logs the value of the current
+//  node and then iterates through its children, calling the function on each child. 
+// The function stops when it encounters a null node.
 
-    if (curr.right) {
-      stack.push(curr.right)
-    }
+function searchDFS(tree, target) {
+  if (tree.value === target) {
+    return true;
+  }
 
-    if (curr.left) {
-      stack.push(curr.left)
+  for (let i = 0; i < tree.children.length; i++) {
```

Comment:
Just use `for (const child of tree.children)`

Reply:
yeah we can use it , its a matter of preference but here it provides much more fine control

Reply:
I don't see why the index would be needed.

Reply:
I have done that